test/api_save_product_test.dart

```diff
@@ -398,6 +398,82 @@ Like that:
       }
     }, skip: 'Works randomly');
 
+    test('Test saving food with PerSize.serving Nutrients', () async {
+      const User USER = TestConstants.TEST_USER;
+      const double ENERGY = 365;
+      const double CARBOHYDRATES = 12;
+      const double PROTEINS = 6;
+      const double FAT = 0.1;
+      const String BARCODE = '01234567891';
+      const String PRODUCT_NAME = 'Test Food';
+      final String nutrimentDataPer = PerSize.serving.offTag;
+
+      const PerSize perSize = PerSize.serving;
+      for (int i = 2; i >= 0; i--) {
```

Comment:
I know you probably just copied that 2 down to 0 loop elsewhere, but that's very strange.
Please test if something less exotic like 0 to 2 works: if so fix this line and the other line in this file with the 2 to 0.

Reply:
Yeah I copied the whole test. I will rewrite it to be more applicable.

Reply:
Flipped that for you!